### PR TITLE
Timeslide injections

### PR DESCRIPTION
**Pull request by EthanMarx**
Library and project for generating timeslides and injections on top of them.

### CODE REVIEW

File: libs/io/bbhnet/io/timeslides.py [hunk 1]
```diff
@@ -342,7 +342,33 @@ def shift(self):
 
     def __post_init__(self):
         self.root = Path(self.root)
+        segment = None
```
Comment (alecgunny) [inline libs/io/bbhnet/io/timeslides.py:345]: I would either call `self.update()` here, or call `self.__post_init__()` in `self.update`, but calling the same code twice will just make maintaining/updating trickier

**Review by alecgunny — CHANGES_REQUESTED** [review]
Only biggish thing for me is the duplicate code between `TimeSlide.update` and `TimeSlide.__post_init__`, but otherwise looks great

File: projects/sandbox/timeslide_injections/timeslide_injections.py [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+    for shifts in timeslides:
+
+        # TODO: might be overly complex naming,
+        # but wanted to attempt to generalize to multi ifo
+        root = outdir / f"dt-{'-'.join(map(str,shifts))}"
+
+        # make root and timeslide directories
+        root.mkdir(exist_ok=True, parents=True)
+        Path(root / "injection").mkdir(exist_ok=True, parents=True)
+        Path(root / "background").mkdir(exist_ok=True, parents=True)
```
[the remark this answers is not in the document]
Comment (EthanMarx) [reply projects/sandbox/timeslide_injections/timeslide_injections.py:190]: Should probably just put a `.mkdir(exist_ok=True)` in the TimeSlide __post_init__ , unless there was a specific reason you didnt

[the remark this answers is not in the document]
Comment (alecgunny) [reply projects/sandbox/timeslide_injections/timeslide_injections.py]: Ahhhh great point. I actually don't know if I think we ought to `mkdir` in `__post_init__`. If I was a user and I specified a bad path I would want to know, not have it silently made. There are a few different ways I can imagine to handle this,
- a `create` kwarg that handles whether to do this
- using a `.create` classmethod that will either initialize an existing timeslide or create the dir if doesn't exist
- Don't do `update` in `__post_init__` at all and make users call it explicitly

None of these, however, are the purview of this PR, so we can resolve this. If this comes up again in the future we can think about these more.

[the remark this answers is not in the document]
Comment (EthanMarx) [reply projects/sandbox/timeslide_injections/timeslide_injections.py]: Went with the class method. Let me know how it looks.

**Review by alecgunny — APPROVED** [review]

**Comment by alecgunny** [comment]
Will partially address #92 by constraining numpy in `bbhnet.base`. In a separate PR I'll go through and update the export project's lockfile and ensure a fresh build works.

**Pull request merged**